hummingbot/connector/exchange/kucoin/kucoin_exchange.pyx

```diff
@@ -389,17 +393,19 @@ cdef class KucoinExchange(ExchangeBase):
         if data is not None:
             data = json.dumps(data)
 
+        if not throttler_passthrough:
+            context = self._throttler.execute_task(limit_id)
+            await context.__aenter__()
         if method == "get":
-            async with self._throttler.execute_task(limit_id):
-                response = await client.get(url, params=params, data=data, headers=headers)
+            response = await client.get(url, params=params, data=data, headers=headers)
         elif method == "post":
-            async with self._throttler.execute_task(limit_id):
-                response = await client.post(url, params=params, data=data, headers=headers)
+            response = await client.post(url, params=params, data=data, headers=headers)
         elif method == "delete":
-            async with self._throttler.execute_task(limit_id):
-                response = await client.delete(url, headers=headers)
+            response = await client.delete(url, headers=headers)
         else:
             response = False
+        if not throttler_passthrough:
+            await context.__aexit__(None, None, None)
```

Comment:
We should not invoke the enter and exit codes like this. The execution using with takes care of, between other things, ensuring the exit code is always executed. We should not break encapsulation like this.

Reply:
ditto